File: CS/MessageBoxServices/XtraMessageBoxServiceUserControl.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DevExpress.Utils.MVVM;
using DevExpress.XtraEditors;
using System;
using System.Windows.Forms;

namespace DxSampleMessageBoxServices {
    public partial class XtraMessageBoxServiceUserControl : XtraUserControl {
        public XtraMessageBoxServiceUserControl() {
            InitializeComponent();
            #region SetUp
            MVVMContext mvvmContext = new MVVMContext();
            mvvmContext.ContainerControl = this;
        
            SimpleButton commandButton = new SimpleButton();
            commandButton.Text = "Execute Command";
            commandButton.Dock = DockStyle.Top;
            commandButton.Parent = this;
            #endregion SetUp

            #region #xtraMessageBoxService
            // Force use the XtraMessageBoxService
            MVVMContext.RegisterXtraMessageBoxService();
            //
            mvvmContext.ViewModelType = typeof(SayHelloViewModel);
            // UI binding for button
            mvvmContext.BindCommand<SayHelloViewModel>(commandButton, x => x.SayHello());
            #endregion #xtraMessageBoxService
        }
        #region CleanUp
        void OnDisposing() {
            MVVMContext.RegisterXtraMessageBoxService();
            var context = MVVMContext.FromControl(this);
            if(context != null) context.Dispose();
        }
        #endregion CleanUp
        public class SayHelloViewModel {
            protected IMessageBoxService MessageBoxService {
                // using the GetService<> extension method for obtaining service instance
                get { return this.GetService<IMessageBoxService>(); }
            }
            public void SayHello() {
                // using the MessageBoxService.ShowMessage method
                if(MessageBoxService.ShowMessage("Hello, buddy! Have a nice day!", "Greeting", MessageButton.OK, MessageIcon.Information) == MessageResult.OK) {
                    // do something
                }
            }
        }
    }
}

File: VB/MessageBoxServices/XtraMessageBoxServiceUserControl.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Utils.MVVM
Imports DevExpress.XtraEditors
Imports System
Imports System.Windows.Forms

Namespace DxSampleMessageBoxServices
    Partial Public Class XtraMessageBoxServiceUserControl
        Inherits DevExpress.XtraEditors.XtraUserControl
        Public Sub New() 
            InitializeComponent()
            Dim mvvmContext As New MVVMContext()
            mvvmContext.ContainerControl = Me
        
            Dim commandButton As New SimpleButton()
            commandButton.Text = "Execute Command"
            commandButton.Dock = DockStyle.Top
            commandButton.Parent = Me

            ' Force use the XtraMessageBoxService
            MVVMContext.RegisterXtraMessageBoxService()
            '
            mvvmContext.ViewModelType = GetType(SayHelloViewModel)
            ' UI binding for button
            mvvmContext.BindCommand(Of SayHelloViewModel)(commandButton, Sub(x) x.SayHello())
        End Sub
         Private Sub OnDisposing()
            MVVMContext.RegisterXtraMessageBoxService()
            Dim context = MVVMContext.FromControl(Me)
            If context IsNot Nothing Then
                context.Dispose()
            End If
        End Sub
        Public Class SayHelloViewModel
            Protected ReadOnly Property MessageBoxService() As IMessageBoxService
                ' using the GetService<> extension method for obtaining service instance
                Get
                    Return Me.GetService(Of IMessageBoxService)()
                End Get
            End Property
            Public Sub SayHello()
                ' using the MessageBoxService.ShowMessage method
                If MessageBoxService.ShowMessage("Hello, buddy! Have a nice day!", "Greeting", MessageButton.OK, MessageIcon.Information) = MessageResult.OK Then
                    ' do something
                End If
            End Sub
        End Class
    	End Class
End Namespace